### PR TITLE
Svelte: Add v5 stories to CLI templates

### PR DESCRIPTION
**Pull request by JReinhold**
Follow-up to https://github.com/storybookjs/storybook/pull/29323

## What I did

Added Svelte 5 templates for the usual Button, Header and Page templates, for JS and TS.

Note 1: The `ts-3-8` and `ts-4-9` directories are identical, in Svelte CSF there's no difference
Note 2: Only the components differ between the JS and TS templates, the story files are identical.

#### The changes in this PR are covered in the following automated tests:
- [ ] stories
- [ ] unit tests
- [ ] integration tests
- [ ] end-to-end tests

#### Manual testing

What I did:
- [x] Initialize the `sveltekit-prerelease-ts` sandbox and verify that new stories are added & it runs
- [x] Initialize the `sveltekit-skeleton-ts` sandbox and verify that old stories are added & it runs

### Documentation

- [ ] Add or update documentation reflecting your changes
- [ ] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

## Checklist for Maintainers

- [ ] When this PR is ready for testing, make sure to add `ci:normal`, `ci:merged` or `ci:daily` GH label to it to run a specific set of sandboxes. The particular set of sandboxes can be found in `code/lib/cli/src/sandbox-templates.ts`
- [ ] Make sure this PR contains **one** of the labels below:
   <details>
     <summary>Available labels</summary>

     - `bug`: Internal changes that fixes incorrect behavior.
     - `maintenance`: User-facing maintenance tasks.
     - `dependencies`: Upgrading (sometimes downgrading) dependencies.
     - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
     - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
     - `documentation`: Documentation **only** changes. Will not show up in release changelog.
     - `feature request`: Introducing a new feature.
     - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
     - `other`: Changes that don't fit in the above categories.
   
   </details>

### 🦋 Canary release

This pull request has been released as version `0.0.0-pr-29382-sha-f608e224`. Try it out in a new sandbox by running `npx storybook@0.0.0-pr-29382-sha-f608e224 sandbox` or in an existing project with `npx storybook@0.0.0-pr-29382-sha-f608e224 upgrade`.
<details>
<summary>More information</summary>

| | |
| --- | --- |
| **Published version** | [`0.0.0-pr-29382-sha-f608e224`](https://npmjs.com/package/storybook/v/0.0.0-pr-29382-sha-f608e224) |
| **Triggered by** | @JReinhold |
| **Repository** | [storybookjs/storybook](https://github.com/storybookjs/storybook) |
| **Branch** | [`svelte-5-stories`](https://github.com/storybookjs/storybook/tree/svelte-5-stories) |
| **Commit** | [`f608e224`](https://github.com/storybookjs/storybook/commit/f608e2240dfd11c6ef29c6b05adfa3533a920752) |
| **Datetime** | Thu Oct 17 12:16:59 UTC 2024 (`1729167419`) |
| **Workflow run** | [11384739899](https://github.com/storybookjs/storybook/actions/runs/11384739899) |

To request a new release of this pull request, mention the `@storybookjs/core` team.

_core team members can create a new canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=29382`_
</details>

| name | before | after | diff | z | % |
| ---- | ------ | ----- | ---- | - | - |
| createSize |  0 B | 0 B | 0 B | - | - |
| generateSize |  78.7 MB | 78.7 MB | 10.6 kB | **3.99** | 0% |
| initSize |  147 MB | 147 MB | 13.2 kB | -0.33 | 0% |
| diffSize |  68.3 MB | 68.3 MB | 2.55 kB | -0.35 | 0% |
| buildSize |  6.79 MB | 6.79 MB | 0 B | 0.34 | 0% |
| buildSbAddonsSize |  1.5 MB | 1.5 MB | 0 B | -0.6 | 0% |
| buildSbCommonSize |  195 kB | 195 kB | 0 B | - | 0% |
| buildSbManagerSize |  1.83 MB | 1.83 MB | 0 B | 1.11 | 0% |
| buildSbPreviewSize |  270 kB | 270 kB | 0 B | **-1.53** | 0% |
| buildStaticSize |  0 B | 0 B | 0 B | - | - |
| buildPrebuildSize |  3.8 MB | 3.8 MB | 0 B | 0.54 | 0% |
| buildPreviewSize |  2.99 MB | 2.99 MB | 0 B | 0.32 | 0% |
| testBuildSize |  0 B | 0 B | 0 B | - | - |
| testBuildSbAddonsSize |  0 B | 0 B | 0 B | - | - |
| testBuildSbCommonSize |  0 B | 0 B | 0 B | - | - |
| testBuildSbManagerSize |  0 B | 0 B | 0 B | - | - |
| testBuildSbPreviewSize |  0 B | 0 B | 0 B | - | - |
| testBuildStaticSize |  0 B | 0 B | 0 B | - | - |
| testBuildPrebuildSize |  0 B | 0 B | 0 B | - | - |
| testBuildPreviewSize |  0 B | 0 B | 0 B | - | - |

| name | before | after | diff | z | % |
| ---- | ------ | ----- | ---- | - | - |
| createTime |  8.4s | 7.7s | -717ms | -0.59 | -9.3% |
| generateTime |  27.6s | 19.7s | -7s -956ms | -0.57 | -40.3% |
| initTime |  18.2s | 12.8s | -5s -420ms | -0.63 | -42.2% |
| buildTime |  8.2s | 8.7s | 518ms | -0.18 | 5.9% |
| testBuildTime |  0ms | 0ms | 0ms | - | - |
| devPreviewResponsive |  7.3s | 5.4s | -1s -859ms | **-1.25** | 🔰-34% |
| devManagerResponsive |  4.4s | 3.5s | -825ms | **-1.3** | 🔰-23% |
| devManagerHeaderVisible |  574ms | 534ms | -40ms | -0.78 | -7.5% |
| devManagerIndexVisible |  603ms | 564ms | -39ms | -0.78 | -6.9% |
| devStoryVisibleUncached |  1.1s | 856ms | -331ms | -0.73 | -38.7% |
| devStoryVisible |  601ms | 563ms | -38ms | -0.8 | -6.7% |
| devAutodocsVisible |  404ms | 449ms | 45ms | -0.97 | 10% |
| devMDXVisible |  542ms | 423ms | -119ms | **-1.49** | 🔰-28.1% |
| buildManagerHeaderVisible |  593ms | 508ms | -85ms | -0.5 | -16.7% |
| buildManagerIndexVisible |  622ms | 520ms | -102ms | -0.49 | -19.6% |
| buildStoryVisible |  623ms | 577ms | -46ms | -0.41 | -8% |
| buildAutodocsVisible |  505ms | 438ms | -67ms | -0.71 | -15.3% |
| buildMDXVisible |  511ms | 402ms | -109ms | -1.01 | -27.1% |

## Greptile Summary

This PR adds Svelte 5 story templates to the Storybook CLI, including new Button, Header, and Page components for both JavaScript and TypeScript.

- Introduced new Svelte 5 templates in `code/renderers/svelte/template/cli` for JS and TS
- Updated `code/core/src/cli/helpers.ts` to support Svelte 5 template selection
- Modified `code/lib/create-storybook/src/generators/SVELTE/index.ts` for simplified Svelte version detection
- Updated Svelte dependency to version ^5.0.0-next.268 in `code/package.json`
- Added new `$props()` and `$state` syntax usage in Svelte 5 components

### CODE REVIEW

**Comment by JReinhold** [comment]
@benmccann could you sanity check my Svelte 5 syntax skills here?

**Comment by benmccann** [comment]
looks good to me!

**Review by JReinhold — COMMENTED** [review]
Thanks for the help @shilman. I tested the canary in Svelte 4 + 5, with TS/JSDoc/JS and everything worked as expected.

I've had to silence some TS errors in our own stories, there's a big risk that our CSF typings for Svelte are slightly incompatible with Svelte 5, but that's not for this PR to solve.

File: code/core/src/cli/helpers.ts [hunk 1]
```diff
@@ -173,20 +173,49 @@ export const frameworkToDefaultBuilder: Record<
...
     // keeping this for backwards compatibility in case community packages are using it
     typescript: 'ts',
     [SupportedLanguage.JAVASCRIPT]: 'js',
     [SupportedLanguage.TYPESCRIPT_3_8]: 'ts-3-8',
     [SupportedLanguage.TYPESCRIPT_4_9]: 'ts-4-9',
   };
+  // FIXME: remove after 9.0
+  if (renderer === 'svelte') {
+    const svelteVersion = await getVersionSafe(packageManager, 'svelte');
+    if (svelteVersion && major(svelteVersion) >= 5) {
```
[the remark this answers is not in the document]
Comment (benmccann) [reply code/core/src/cli/helpers.ts:213]: Bad idea greptile. You don't want Svelte 4 behavior for Svelte 6

**Comment by JReinhold** [comment]
Merging, CI failures are unrelated to this work.

**Pull request merged**